midi/src/lib.rs

```diff
@@ -0,0 +1,19 @@
+//! MIDI message extension for `lv2_atom`.
+//!
+//! This crate adds a new atom type for the `lv2_atom` crate: The `MidiEvent`, a message conformant to the [MIDI specification](https://www.midi.org/specifications-old/item/the-midi-1-0-specification). Due to the one-crate-per-spec policy of the `rust-lv2` project, this relatively small crate isn't integrated into the main atom crate.
+//!
+//! If compiled with the optional `wmidi` dependency, `MidiEvent` will use the `MidiMessage` type defined in by `wmidi` instead of byte slices.
+extern crate lv2_atom as atom;
+extern crate lv2_core as core;
+pub extern crate lv2_midi_sys as sys;
+extern crate lv2_urid as urid;
+
+#[cfg(not(feature = "wmidi"))]
```

Comment:
This can cause trouble: suppose crate `A` uses `rust-lv2` without the `wmidi` feature, then `<MidiEvent as Atom>::ReadHandle` is `&[u8]`. Suppose crate `A` also has a dependency on crate `B`. If crate `B` adds a dependency to `rust-lv2` _with_ the `wmidi` feature, then `cargo` will compile `rust-lv2` only once, _with_ the `wmidi` feature and suddenly, for crate `A`, `<MidiEvent as Atom>::ReadHandle` is no longer `&[u8]`, but `wmidi::MidiMessage`, breaking `A`'s code.

Reply:
I thought that `cargo` would compile that crate twice, once for each feature set, but I tried it and you're right! What do you suppose to do?

Reply:
> What do you suppose to do?

Good question!
`#[cfg(not(feature = "wmidi"))]` should be removed, but the question is then: how to proceed? From the top of my head, I see the following options:
* Make `Atom` generic over `ReadHandle` and have two implementations of `Atom` for `MidiEvent`.
* Instead of one `MidiEvent` have both a `RawMidiEvent` and a `StructuredMidiEvent` (maybe find a better name than this) instead of just `MidiEvent` alone.

Maybe these ideas do not make sense at all.
I would have to dive deeper into the code to see if these make sense and maybe there are other options that are better than the ones I listed above.

Reply:
The second approach would make sense: The `wmidi` feature would enable this additional struct, which would be handled as a completely different atom. Therefore, enabling the feature would not break anything. I'm going to implement that!